tests/test_rich_markup_mode.py

```diff
@@ -38,3 +41,304 @@ def main(arg: str):
+    assert help_start != -1
+    assert result_lines[help_start:arg_start] == lines
+
+
+@pytest.mark.parametrize(
+    "mode,lines",
+    [
+        ("markdown", ["First line", "", "• 1", "• 2", "• 3", ""]),
+        pytest.param(
+            "rich",
+            ["First line", "", "- 1", "", "- 2", "", "- 3", ""],
+            marks=pytest.mark.xfail,
+        ),
+        pytest.param(
+            "none",
+            ["First line", "", "- 1", "", "- 2", "", "- 3", ""],
+            marks=pytest.mark.xfail,
+        ),
+    ],
+)
+def test_markup_mode_bullets_double_newline(mode: str, lines: List[str]):
+    app = typer.Typer(rich_markup_mode=mode)
+
+    @app.command()
+    def main(arg: str):
+        """First line
+
+        - 1
+
+        - 2
+
+        - 3
+        """
+        print(f"Hello {arg}")
+
+    assert app.rich_markup_mode == mode
+
+    result = runner.invoke(app, ["World"])
+    assert "Hello World" in result.stdout
+
+    result = runner.invoke(app, ["--help"])
+    result_lines = [line.strip() for line in result.stdout.split("\n")]
+    assert any(c in result.stdout for c in rounded)
+    help_start = result_lines.index("First line")
+    arg_start = [i for i, row in enumerate(result_lines) if "Arguments" in row][0]
+    assert help_start != -1
+    assert result_lines[help_start:arg_start] == lines
+
+
+@pytest.mark.parametrize(
+    "mode,lines",
+    [
+        ("markdown", ["First line", "", "• 1 - 2   - a   - b - 3", ""]),
+        pytest.param(
+            "rich",
+            ["First line", "", "- 1 - 2  - a  - b - 3", ""],
+            marks=pytest.mark.xfail,
+        ),
+        pytest.param(
+            "none",
+            ["First line", "", "- 1 - 2  - a  - b - 3", ""],
+            marks=pytest.mark.xfail,
+        ),
+    ],
+)
+def test_markup_mode_nested_bullets_single_newline(mode: str, lines: List[str]):
+    app = typer.Typer(rich_markup_mode=mode)
+
+    @app.command()
+    def main(arg: str):
+        """First line
+
+        - 1
+        - 2
+          - a
+          - b
+        - 3
+        """
+        print(f"Hello {arg}")
+
+    assert app.rich_markup_mode == mode
+
+    result = runner.invoke(app, ["World"])
+    assert "Hello World" in result.stdout
+
+    result = runner.invoke(app, ["--help"])
+    result_lines = [line.strip() for line in result.stdout.split("\n")]
+    assert any(c in result.stdout for c in rounded)
+    help_start = result_lines.index("First line")
+    arg_start = [i for i, row in enumerate(result_lines) if "Arguments" in row][0]
+    assert help_start != -1
+    assert result_lines[help_start:arg_start] == lines
+
+
+@pytest.mark.parametrize(
+    "mode,lines",
+    [
+        ("markdown", ["First line", "", "• 1", "• 2", "• a", "• b", "• 3", ""]),
+        pytest.param(
+            "rich",
+            ["First line", "", "- 1", "", "- 2", "", "- a", "", "- b", "", "- 3", ""],
+            marks=pytest.mark.xfail,
+        ),
```

Comment:
This one fails on `master` only because of a missing `""` right after the header